app/src/main/java/org/oppia/app/profile/AdminSettingsDialogFragmentPresenter.kt

```diff
@@ -0,0 +1,73 @@
+package org.oppia.app.profile
+
+import android.app.Dialog
+import android.content.Context
+import android.text.Editable
+import android.text.TextWatcher
+import androidx.appcompat.app.AlertDialog
+import androidx.databinding.DataBindingUtil
+import androidx.fragment.app.Fragment
+import org.oppia.app.R
+import org.oppia.app.databinding.AdminSettingsDialogBinding
+import org.oppia.app.fragment.FragmentScope
+import org.oppia.app.viewmodel.ViewModelProvider
+import javax.inject.Inject
+
+/** The presenter for [AdminAuthActivity]. */
+@FragmentScope
+class AdminSettingsDialogFragmentPresenter @Inject constructor(
+  private val fragment: Fragment,
+  private val viewModelProvider: ViewModelProvider<AdminSettingsViewModel>
+) {
+  private val adminViewModel by lazy {
+    getAdminSettingsViewModel()
+  }
+
+  fun handleOnCreateDialog(routeDialogInterface: ProfileRouteDialogInterface): Dialog {
+    val adminPin = fragment.arguments?.getString(KEY_ADMIN_SETTINGS_PIN)
+    checkNotNull(adminPin) { "Admin Pin must not be null" }
+    val binding: AdminSettingsDialogBinding = DataBindingUtil.inflate(fragment.requireActivity().layoutInflater, R.layout.admin_settings_dialog, null, false)
+    binding.apply {
+      lifecycleOwner = fragment
+      viewModel = adminViewModel
+    }
+
+    binding.inputPin.addTextChangedListener(object: TextWatcher {
+      override fun onTextChanged(confirmPin: CharSequence?, start: Int, before: Int, count: Int) {
+        confirmPin?.let {
+          adminViewModel.errorMessage.set("")
+        }
+      }
+      override fun afterTextChanged(confirmPin: Editable?) {}
+      override fun beforeTextChanged(p0: CharSequence?, start: Int, count: Int, after: Int) {}
+    })
+
+    val dialog =  AlertDialog.Builder(fragment.requireActivity() as Context, R.style.AlertDialogTheme)
+      .setTitle(R.string.admin_settings_heading)
+      .setView(binding.root)
+      .setMessage(R.string.admin_settings_sub)
+      .setPositiveButton(R.string.admin_settings_submit, null)
+      .setNegativeButton(R.string.admin_settings_cancel) { dialog, _ ->
+        dialog.dismiss()
+      }
+      .create()
+
+    dialog.setOnShowListener {
+      dialog.getButton(AlertDialog.BUTTON_POSITIVE).setOnClickListener {
```

Comment:
I'm confused: why aren't we just setting this click listener when setting the positive button in the builder above? This setup seems a bit unusual.

Reply:
https://stackoverflow.com/questions/2620444/how-to-prevent-a-dialog-from-closing-when-a-button-is-clicked

This ensures that the dialog fragment will not be dismissed if the user input is incorrect.

Reply:
Aha. I see you linked to the SO question above--maybe also add a brief comment explaining the logic. Also, I suggest using the shortened SO link for brevity: https://stackoverflow.com/a/7636468 (you can find this using the 'share' button for the desired answer/question that you want to link to).

Reply:
Thanks for the shortened url tip! Updated the comment.